synapse/replication/tcp/protocol.py

```diff
@@ -201,15 +201,23 @@ def send_ping(self):
                 )
                 self.send_error("ping timeout")
 
-    def lineReceived(self, line):
+    def lineReceived(self, line: bytes):
         """Called when we've received a line
         """
         if line.strip() == "":
             # Ignore blank lines
             return
 
-        line = line.decode("utf-8")
-        cmd_name, rest_of_line = line.split(" ", 1)
+        linestr = line.decode("utf-8")
+
+        # split at the first " ", handling one-word commands
+        idx = linestr.index(" ")
+        if idx >= 0:
+            cmd_name = linestr[:idx]
+            rest_of_line = linestr[idx + 1 :]
+        else:
+            cmd_name = linestr
+            rest_of_line = ""
```

Comment:
I would have used `split(maxsplit=1)`, but doesn't seem to really simplify this code at all.

Reply:
well, `split(maxsplit=1)` is what we have at https://github.com/matrix-org/synapse/pull/7239/files#diff-88dc76170973754b5122a253db41d339L212.

indeed this impl feels cumbersome, but I'm not aware of a better way :-S